src/Swift.Bindings/src/Emitter/StringCSharpEmitter/Handler/MethodHandler.cs

```diff
@@ -201,14 +207,21 @@ private void EmitWrapperMethod(IndentedTextWriter writer, MethodEnvironment env)
                     writer.WriteLine($"var self = new SwiftSelf((void*)_payload);");
             }
 
-            // TODO: Add Indirect result marshalling to methods other than constructors
-
-            var returnPrefix = methodDecl.CSSignature.First().SwiftTypeSpec is TupleTypeSpec tupleTypeSpec && tupleTypeSpec.IsEmptyTuple ? "" : "return ";
-            var invokeArguments = env.SignatureHandler.GetPInvokeSignature().CallArgumentsString();
-
-            // Call the PInvoke method
-            writer.WriteLine($"{returnPrefix}{pInvokeName}({invokeArguments});");
+            if (requiresIndirectResult)
+            {
+                writer.WriteLine($"var payload = (SwiftHandle)NativeMemory.Alloc({wrapperSignature.ReturnType}.PayloadSize);");
```

Comment:
Nit: A single write with multiple lines would improve readability.

Reply:
I agree. We discussed that on one of the previous PRs that single write will mess up the indent. I have a task #2891 on my backlog to improve that